imaging/ml/ml_codelab/breast_density_auto_ml.ipynb

```diff
@@ -1,935 +1,972 @@
+   "source": [
+    "Next, we will train the model to perform classification. We will set the training budget to be a maximum of 1hr (but this can be modified below). The cost of using AutoML can be found [here](https://cloud.google.com/vision/automl/pricing). Typically, the longer the model is trained for, the more accurate it will be."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "kUJnoVd8k_hy"
+   },
+   "outputs": [],
+   "source": [
+    "# Name of the model.\n",
+    "model_display_name = \"MY_MODEL_NAME\" # @param\n",
+    "\n",
+    "# Training budget (1 hr).\n",
+    "training_budget = 1 # @param"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "voU4GkP6lWKE"
+   },
+   "outputs": [],
+   "source": [
+    "# Path to request import into AutoML dataset.\n",
+    "path = os.path.join(AUTOML_API_URL, 'projects', project_id, 'locations', location, 'models')\n",
+    "\n",
+    "# Headers (request in JSON format).\n",
+    "headers = {'Content-Type': 'application/json'}\n",
+    "\n",
+    "# Body (encoded in JSON format).\n",
+    "automl_dataset_id = automl_dataset_name.split('/')[-1]\n",
+    "config = {'display_name': model_display_name, 'dataset_id': automl_dataset_id, 'image_classification_model_metadata': {'train_budget': training_budget}}\n",
+    "body = json.dumps(config)\n",
+    "\n",
+    "resp, content = http.request(path, method='POST', headers=headers, body=body)\n",
+    "assert resp.status == 200, 'error creating AutoML model, code: {0}, response: {1}'.format(resp.status, content)\n",
+    "print('Full response:\\n{0}'.format(content.decode()))\n",
+    "\n",
+    "# Record operation_name so we can poll for it later.\n",
+    "response = json.loads(content.decode())\n",
+    "operation_name = response['name']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "EADtTufcnPFf"
+   },
+   "source": [
+    "The output of the previous step is also an [operation](https://cloud.google.com/vision/automl/docs/models#get-operation) that will need to poll the status of. We will poll until the operation's \"done\" field is set to true. This will take a few minutes to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "vsl3iHa2niLo"
+   },
+   "outputs": [],
+   "source": [
+    "path = os.path.join(AUTOML_API_URL, operation_name)\n",
+    "timeout = time.time() + 20*60 # Wait up to 20 minutes.\n",
+    "response = wait_for_operation_completion(path, timeout)\n",
+    "full_model_name = response['response']['name']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!sudo pip3 install google.cloud.automl\n",
```

Comment:
Same thing here. Add a comment and perhaps separate into another block. Inline pip3 installs can be misssed.

Reply:
comment is added